### ISSUE TITLE
`valid-expect` should allow Vitest's `not.resolves` and `not.rejects` chains

### ISSUE THREAD

**Issue by nami8824** [vitest-dev/eslint-plugin-vitest#919]
## Summary

`valid-expect` currently treats some expect modifier combinations in a Jest-like way.

However, Vitest's expect API is Chai-style chainable, so `not.resolves` and `not.rejects` are valid Vitest assertions:

```ts
await expect(Promise.resolve(1)).not.resolves.toBe(2)
await expect(Promise.reject(new Error('error'))).not.rejects.toThrow('other')
```

These chains should not be reported as `modifierUnknown`.

This is different from Jest. In Jest, promise modifiers are exposed as `resolves.not` / `rejects.not`, and `not.resolves` / `not.rejects` are not part of the runtime API.

For example, this fails in Jest:

```ts
await expect(Promise.resolve(1)).not.resolves.toBe(2)
```

with:

```txt
TypeError: Cannot read properties of undefined (reading 'toBe')
```

The same assertion passes in Vitest.

## Reproduction

This should be valid:

```ts
await expect(Promise.resolve(1)).not.resolves.toBe(2)
```

This should be reported as missing `await` / `return`:

```ts
expect(Promise.resolve(1)).not.resolves.toBe(2)
```

Currently, the unawaited case can be reported as `modifierUnknown` instead of being treated as a valid async assertion chain.

## Expected behavior

`valid-expect` should treat `not.resolves` and `not.rejects` as valid Vitest expect chains.

They should still be treated as async assertions because `resolves` / `rejects` is present.

## Notes

This should intentionally differ from `eslint-plugin-jest`, because Jest and Vitest expose different expect chain APIs.

This issue does not cover duplicate `not` chains such as `not.not`, since that is more about lint policy than promise modifier compatibility.
